.pre-commit-config.yaml

```diff
@@ -13,27 +13,11 @@ repos:
   - id: mixed-line-ending
   - id: trailing-whitespace
 
-- repo: https://github.com/asottile/pyupgrade
-  rev: v3.3.1
+- repo: https://github.com/charliermarsh/ruff-pre-commit
+  rev: v0.0.226
   hooks:
-  - id: pyupgrade
-    args: ["--py37-plus"]
-    exclude: ^cibuildwheel/resources/.*py$
-
-# Autoremoves unused imports
-- repo: https://github.com/hadialqattan/pycln
-  rev: v2.1.2
-  hooks:
-  - id: pycln
-    args: [--all]
-    stages: [manual]
-
-- repo: https://github.com/PyCQA/isort
-  rev: 5.11.4
-  hooks:
-  - id: isort
-    args: ["-a", "from __future__ import annotations"]
-    exclude: ^cibuildwheel/resources/.*py$
+  - id: ruff
+    args: ["--fix"]
```

Comment:
Maybe move this after black. 
Then there will be a lower probability that some formatting rule would be reported.

Reply:
Okay. Though Ruff has auto-fixers too.

Reply:
Ruff after black caused a double fix: https://results.pre-commit.ci/run/github/525643569/1674247644.l9WEjn23TbWUFrKVbXv-2w (The other way might too, but this way certainly did)

Reply:
It's because the fix rewrite created unformatted code. I think Ruff -> Black might be a better order.